internal/speakerlist/speakerlist.go

```diff
@@ -0,0 +1,215 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package speakerlist
+
+import (
+	"crypto/sha256"
+	"errors"
+	golog "log"
+	"reflect"
+	"strconv"
+	"sync"
+	"time"
+
+	"go.universe.tf/metallb/internal/k8s"
+	v1 "k8s.io/api/core/v1"
+
+	gokitlog "github.com/go-kit/kit/log"
+	"github.com/hashicorp/memberlist"
+)
+
+// SpeakerList gives you the list of healthy speaker
+type SpeakerList struct {
+	sync.Mutex
+	l           gokitlog.Logger
+	client      *k8s.Client
+	mList       *memberlist.Memberlist
+	k8sSpeakers map[string]string
+	eventCh     chan memberlist.NodeEvent
+	stopCh      chan struct{}
+}
+
+// New creates a new SpeakerList
+func New(logger gokitlog.Logger, nodeName, bindAddr, bindPort, secret string, stopCh chan struct{}) (*SpeakerList, error) {
+	sl := SpeakerList{
```

Comment:
Maybe mention in a comment that client is set at `Start()`? Because we are returning a struct that has some fields if some params are enabled, and some others if not but is never complete. I think it might be easier to follow if we just add a simple comment (thinking in future me, that will forget about all of that :))

Reply:
`Start()` is gone (I've replaced the k8s client with a chan)